accepted/capability-checks/capability-checks.md

```diff
@@ -0,0 +1,234 @@
+# Capability APIs
+
+**PM** [Immo Landwerth](https://github.com/terrajobst)
+
+Thoughts I shared on Twitter:
+
+* This morning it happened. I had an epiphany under the shower to make
+  platform-checks much better via a generalized approach for capability APIs.
+  Basically, instead of doing actual platform checks, the analyzer would push you
+  call capability APIs.
+* The nice thing is that this also works for features that aren't OS specific,
+  but more of a combination of runtime and environment, such as whether you can
+  JIT.
+* Imagine in web assembly where we discover certain features wouldn't work. It's
+  hard to annotate an API with "works everywhere but here" in a fashion that
+  doesn't break consumers when the environment changes.
+
+## Scenarios and User Experience
+
+## Requirements
+
+### Goals
+
+### Non-Goals
+
+## Design
+
+### Open Questions
+
+#### How can we reduce the noise?
+
+Basically, we should make sure that people building applications don't have to
+perform capability checks if the answer is statically known. For example,
+registry will work on Windows, so folks building a WinForms/WPF app shouldn't
+have to suppress them. One option is add typical capabilities to the assembly,
+but that's messy. Another option is tag capabilities with platform annotations
+that basically say "on this platform, you don't have to ask for capability
+checks". Then it's only a matter of flowing the platform information from the
+build into the analyzer.
+
+#### Should this analyzer be-opt in?
+
+This might be useful for .NET Standard library developers and much less useful
+for people building .NET Core apps. We might also want different defaults.
+
+#### Should we provide an assert-style attribute?
+
+This would behave the same was tagging the assemblies, modules, types, or
+members with the `[RequiredXxxCapability]` but without requiring callers to
+check it. It's basically a suppression for a specific capability.
```

Comment:
If it looks like suppression and behaves like suppression -> it might as well be a suppression 😄  - Why would not `SuppressMessage` (or the new `UnconditionalSuppressMessage`) work here? Maybe I don't understand the "assertion" pattern.